libnetwork/pasta/pasta_linux.go

```diff
@@ -205,8 +203,11 @@ func createPastaArgs(opts *SetupOptions) ([]string, []string, error) {
 	if noUDPNamespacePorts {
 		cmdArgs = append(cmdArgs, "-U", "none")
 	}
-	if noMapGW {
+	if noMapGWIndex < 0 {
 		cmdArgs = append(cmdArgs, "--no-map-gw")
+	} else {
+		// not an actual pasta(1) option so we have to trim it out
+		cmdArgs = append(cmdArgs[:noMapGWIndex], cmdArgs[noMapGWIndex+1:]...)
 	}
```

Comment:
This would fail if you somehow end up with `--map-gw` being passed in twice.  (If it's e.g. a default option in `/etc/containers/containers.conf` _and_ specified explicitly on the command line.)

(Additionally, `noMapGWIndex` feels a bit confusing since it's actually the index of the `--map-gw` option.)

Reply:
Wow, timing. :)

Reply:
yes it will not work for more than one option, but well most pasta options cannot be set more than once anyway so I did not consider this a problem worth fixing.

I guess it it could be regression as the previous code would have removed all --map-gw so maybe I should juts fix it properly even if this means more complex code.

Reply:
I'm a bit confused, it seems to me like pretty much _all_ pasta arguments can be passed multiple times?

I'd be very surprised to have things break just because the same argument was passed twice, especially if one of the times was in a place that I do not control.

Reply:
```
$ pasta --quiet --quiet  true
Multiple --quiet options given
```

Yes many options work fine if given multiple times but others do not with pasta so it is hard to tell which ones work and don't.
But given --map-gw is our own parsing I agree that it should be fixed

Reply:
https://github.com/containers/common/pull/1968

Reply:
> I'd be very surprised to have things break just because the same argument was passed twice

About this in pasta itself: we had a [discussion](https://archives.passt.top/passt-dev/895dae7d-3e61-4ef7-829a-87966ab0bb3a@redhat.com/) around that a while ago, and while silently ignoring all duplicate options could be convenient in most cases, it could hide subtle bugs in users such as Podman. So we concluded we wouldn't do that, at least not in a generalised way.